src/view/xwayland.cpp

```diff
@@ -49,6 +52,13 @@ class wayfire_xwayland_view_base : public wf::wlr_view_t
         on_destroy.set_callback([&] (void*) { destroy(); });
         on_configure.set_callback([&] (void* data) {
             auto ev = static_cast<wlr_xwayland_surface_configure_event*> (data);
+            if ((ev->mask & XCB_CONFIG_WINDOW_X) && (ev->mask & XCB_CONFIG_WINDOW_Y))
+            {
+                self_positioned = true;
+            } else {
```

Comment:
Why do we need the else here? Wlroots will already set x and y to `xsurface->x/y`

Reply:
This is needed or else xterm is placed at 0,0 instead of 100,100 without place enabled.

Reply:
Okay but this is still wrong because ev->x/y is supposed to be global coordinates, and geometry.x/y is geometry relative to the view output.

Reply:
Should be fixed now.